include/sdf/parser.hh

```diff
@@ -333,6 +333,17 @@ namespace sdf
   SDFORMAT_VISIBLE
   bool convertFile(const std::string &_filename, const std::string &_version,
                    const ParserConfig &_config, SDFPtr _sdf);
+  /// \brief Convert an SDF file to a specific SDF version.
+  /// \param[in] _filename Name of the SDF file to convert.
+  /// \param[in] _version Version to convert _filename to.
+  /// \param[in] _config Custom parser configuration
+  /// \param[out] _sdf Pointer to the converted SDF document.
+  /// \param[out] _errors Vector of errors.
+  /// \return True on success.
+  SDFORMAT_VISIBLE
+  bool convertFile(const std::string &_filename, const std::string &_version,
```

Comment:
Can you add a TODO to deprecate the other overloads?

Reply:
You mean to deprecate all overloads that we are creating for the error reporting right? where would you say we should put this TODO?

Reply:
Good question. We should definitely create a ticket for it, but I'm not sure we'll remember to check, so I suggest putting a TODO in at least one of these files (probably parser.hh since it's more frequently updated/looked at)

Reply:
Added an issue [here](https://github.com/gazebosim/sdformat/issues/1186) and the TODO in https://github.com/gazebosim/sdformat/pull/1123/commits/5f91d45a201662332d045ab845cda5ddea827e96.